iothub_client/tests/common_dt_e2e/iothubclient_common_dt_e2e.c

```diff
@@ -541,6 +541,23 @@ static void device_desired_deinit(DEVICE_DESIRED_DATA *device)
     }
 }
 
+static void device_desired_reset(DEVICE_DESIRED_DATA *device)
```

Comment:
Nit - consider this as `device_receivedcallback_reset` or some such as I think that captures more how variable is named & callback is set.  Your call either way as its minor.